sparse/coo/core.py

```diff
@@ -712,9 +722,16 @@ def reduce(self, method, axis=(0,), keepdims=False, **kwargs):
                            np.prod([self.shape[d] for d in axis], dtype=np.intp)))
 
             result, inv_idx, counts = _grouped_reduce(a.data, a.coords[0], method, **kwargs)
-            missing_counts = counts != a.shape[1]
-            result[missing_counts] = method(result[missing_counts],
-                                            self.fill_value, **kwargs)
+
+            result_fill_value = self.fill_value
+
+            if reduce_super_ufunc is None:
+                missing_counts = counts != a.shape[1]
+                result[missing_counts] = method(result[missing_counts],
+                                                self.fill_value, **kwargs)
+            else:
+                result = method(result, reduce_super_ufunc(self.fill_value, a.shape[1] - counts)).astype(result.dtype)
```

Comment:
`counts.dtype` is almost definitely `int64` or `uint64`, plenty. Using `astype` to ensure not too big a type, as `result.dtype` is the dtype of the reduced array.